firmware_mod/scripts/telegram-bot-daemon.sh

```diff
@@ -43,13 +43,13 @@ imageAlerts() {
 
 respond() {
```

Comment:
What a coincidence! I was about to propose a similar pull request, because I implemented groups handling on my camera this morning. If _chatId_ is a negative number, this means it comes from a group (command arrives as **/command**@_botname_).

So, I used shell's native substring handling in order not to change individual case comparison patterns. I put following two statements in the beginning of the _respond()_ function, just before the case statement on line 45: 

```shell
cmd=$1
[ $chatId -lt 0 ] && cmd=${1%%@*}
```
Then the argument **$1** needs to be replaced with **$cmd** (lines 45 and 53).

As an added bonus, my proposed solution will correctly handle situations where commands coming from group chats appear to be substrings of another command (for example /mem will not be matched, if command sent was **/mem**bers).

Reply:
Can you make a pull request for your solution?

Reply:
Sure!

I was out and about in the mountains with no internet access.. so back to reality now, here's the pull request:

[Pull Request #1](https://github.com/m-pavel/Xiaomi-Dafang-Hacks/pull/1)